File: androidApp/features/home/src/main/kotlin/com.velkonost.getbetter.android.features.home/HomeScreen.kt
package com.velkonost.getbetter.android.features.home

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.material3.Button
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.unit.sp
import com.velkonost.getbetter.shared.features.home.presentation.HomeViewModel
import com.velkonost.getbetter.shared.features.home.presentation.contracts.NavigateToDetails

@Composable
internal fun HomeScreen(
    viewModel: HomeViewModel,
    modifier: Modifier = Modifier
) {
    Box(
        modifier = modifier
            .fillMaxSize()
            .background(color = Color.Transparent)
    ) {
        Column {

            Text(text = "123", color = Color.Red, fontSize = 30.sp)

            Button(onClick = {
                viewModel.dispatch(NavigateToDetails)
            }) {
                Text(text = "Click me mthfcr", color = Color.Green, fontSize = 20.sp)
            }
        }

    }

}

File: iosApp/iosApp/Features/Home/HomeScreen.swift
//
//  HomeScreen.swift
//  iosApp
//
//  Created by velkonost on 06.09.2023.
//  Copyright © 2023 orgName. All rights reserved.
//

import Foundation
import SharedSDK
import SwiftUI
import KMMViewModelSwiftUI

struct HomeScreen: View {
    
    @ObservedViewModel var viewModel: HomeViewModel
    
    private let safeAreaInsetTop = UIApplication.shared.windows.first?.safeAreaInsets.top ?? 0
    
    var body: some View {
        ZStack(alignment: .topLeading) {
            
            Color.red
            
            VStack {
                Text("123")
                    .foregroundColor(Color.red)
                    .font(.largeTitle)
                
                Button(action: {
                    viewModel.dispatch(action: NavigateToDetails())
                }) {
                    Text("click me mthfcr")
                        .foregroundColor(Color.green)
                        .padding(10)
                        .font(.title2)
                }
                .background(Color.blue)
                .cornerRadius(30)
                .padding(.init(top: 20, leading: .zero, bottom: .zero, trailing: .zero))
                
//                Image("\(SharedR.shared.image.ic_test)")
                
//                Image(uiImage: SharedR.images().bg_paw_print_loaded.toUIImage()!)
//                Image(name: SharedR.shared.image.ic_test)
            }
            
        }
    }
}

//struct HomeScreenPreview: PreviewProvider {
//    static var previews: some View {
//        HomeScreen()
//    }
//}
